src/devrules/notifications/channels/slack.py

```
@@ -0,0 +1,114 @@
+
+        req = urllib.request.Request(
+            url="https://slack.com/api/chat.postMessage",
+            data=data,
+            headers={
+                "Authorization": f"Bearer {self.token}",
+                "Content-Type": "application/json",
+            },
+            method="POST",
+        )
+
+        with urllib.request.urlopen(req) as response:
```

Ruff (0.14.10)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)